packages/@aws-cdk/aws-ec2/lib/instance-types.ts

```diff
@@ -362,6 +362,16 @@ export enum InstanceClass {
    */
   I3EN = 'i3en',
 
+  /**
+   * Storage optimized instances powered by Graviton2 processor, 4th generation
+   */
+  IM4GN = 'im4gn',
+
+  /**
+   * Storage optimized instances powered by Graviton2 processor, 4th generation
+   */
+  IS4GEN = 'is4gen',
```

Comment:
Could you please add symbolic names for both enums, too? Thanks.

Reply:
Can you suggest one of IS4GEN?

Reply:
Based on the [product page](https://aws.amazon.com/de/ec2/instance-types/i4g/), maybe we could use `STORAGE4_HIGH_PERFORMANCE =  'im4gn',` and `STORAGE4_LOW_COST =  'is4gen',`?

Unfortunately, I can't find any further information on how the abbreviations are composed. We can also wait for further feedback or ideas.

Reply:
@njlynch wdyt?

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#instance-hardware-specs

> Instance type names combine the instance family, generation, and size. They can also indicate additional capabilities, such as:
> * a – AMD processors
> * g – AWS Graviton processors
> * i – Intel processors
> * d – Instance store volumes
> * n – Network optimization
> * b – Block storage optimization
> * e – Extra storage or memory
> * z – High frequency

So these instances are storage-optimized, Graviton-driven, and network-optimized, with the latter also having extra storage/memory. That's a mouthful to stuff into an enum name.

`STORAGE4_GRAVITON_NETWORK_OPTIMIZED` and ``STORAGE4_GRAVITON_NETWORK_STORAGE_OPTIMIZED` would be some suggestions, but I don't love them. Open to counter proposals.

Reply:
Sounds good to me @njlynch. PR updated.